dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -152,6 +158,26 @@ public TrackerBundleReport commit( TrackerBundle bundle )
         return bundleReport;
     }
 
+    @Override
+    public void postCommit( TrackerBundle bundle )
+    {
+        updateTeisLastUpdated( bundle );
+    }
+
+    private void updateTeisLastUpdated( TrackerBundle bundle )
+    {
+        if ( Optional.ofNullable( bundle.getUpdatedTeis() ).filter( ut -> !ut.isEmpty() ).isPresent() )
```

Comment:
Is this Optional.ofNullable really needed?
updatedTeis has a default value and values are just added to it

Reply:
we should always check nullable, it is no harm and it avoids null pointer if software is changed in the future